apps/api/v1/pages/api/credential-sync/_post.ts

```diff
@@ -88,14 +90,11 @@ async function handler(req: NextApiRequest) {
 
   const appMetadata = appStoreMetadata[app.dirName as keyof typeof appStoreMetadata];
 
-  const createdcredential = await prisma.credential.create({
-    data: {
-      userId,
-      appId: appSlug,
-      key,
-      type: appMetadata.type,
-    },
-    select: credentialForCalendarServiceSelect,
+  const createdcredential = await CredentialRepository.create({
```

Comment:
The Prisma.JsonNull handling for key is also a solid choice for cleaner code. Nicely done!